be/src/pipeline/exec/set_source_operator.cpp

```diff
@@ -139,9 +138,21 @@ Status SetSourceOperatorX<is_intersect>::_get_data_in_hashtable(
                 _add_result_columns(local_state, value, block_size);
             }
         }
+    };
+
+    auto& iter = hash_table_ctx.iterator;
+    for (; iter != hash_table_ctx.hash_table->end() && block_size < batch_size; ++iter) {
+        add_result(iter->get_second());
     }
 
     *eos = iter == hash_table_ctx.hash_table->end();
+    if (*eos && hash_table_ctx.hash_table->has_null_key_data()) {
+        auto value = hash_table_ctx.hash_table->template get_null_key_data<RowRefListWithFlags>();
```

Comment:
should here need copy? why here need check type?

Reply:
has_null_key_data return true means hash table is `DataWithNullKey`, return type of other types of hash tables(except DataWithNullKey) is not RowRefListWithFlags